src/contracts/core/DelegationManager.sol

```diff
@@ -535,137 +572,151 @@ contract DelegationManager is Initializable, OwnableUpgradeable, Pausable, Deleg
-                        });
-                    }
                 } else {
-                    strategyManager.addShares(msg.sender, tokens[i], withdrawal.strategies[i], withdrawal.shares[i]);
-                    // Similar to `isDelegated` logic
-                    if (currentOperator != address(0)) {
-                        _increaseOperatorShares({
-                            operator: currentOperator,
-                            // the 'staker' here is the address receiving new shares
-                            staker: msg.sender,
-                            strategy: withdrawal.strategies[i],
-                            shares: withdrawal.shares[i]
-                        });
-                    }
+                    strategyManager.withdrawSharesAsTokens(msg.sender, withdrawal.strategies[i], shares, tokens[i]);
+                }
+            } else {
+                // TODO: unfortunately there doesn't seem to be a good way to avoid scaling down then up
+                uint256 nonNormalizedShares = SlashingAccountingUtils.denormalize({
+                    shares: shares,
+                    scalingFactor: slasher.shareScalingFactor(currentOperator, withdrawal.strategies[i])
+                });
+
+                // TODO DONE?: figure out how to award correct number of shares back when a stale withdrawal is completed
+                // Award shares back in StrategyManager/EigenPodManager. If withdrawer is delegated, increase the shares delegated to the operator
+                // When awarding podOwnerShares in EigenPodManager, we need to be sure to only give them back to the original podOwner.
+                // Other strategy shares can + will be awarded to the withdrawer.
+                if (withdrawal.strategies[i] == beaconChainETHStrategy) {
+                    // Update shares amount depending upon the returned value.
+                    // The return value will be lower than the input value in the case where the staker has an existing share deficit
+                    nonNormalizedShares = eigenPodManager.addShares({
+                        podOwner: withdrawal.staker,
+                        shares: nonNormalizedShares
+                    });
+                } else {
+                    strategyManager.addShares(msg.sender, tokens[i], withdrawal.strategies[i], nonNormalizedShares);
+                }
+
+                // Similar to `isDelegated` logic
+                if (currentOperator != address(0)) {
+                    _increaseOperatorShares({
+                        operator: currentOperator,
+                        // the 'staker' here is the address receiving new shares
+                        staker: withdrawal.staker,
+                        strategy: withdrawal.strategies[i],
+                        nonNormalizedShares: nonNormalizedShares
+                    });
                 }
-                unchecked { ++i; }
             }
+            unchecked { ++i; }
```

Comment:
Let's put this into the `for` loop itself rather than incrementing way down here.

We've been bitten by this before - and this method is getting massive now...

Reply:
See #609